### PR TITLE
Add recommendations for collections

### PR DESCRIPTION
**Pull request by jillr**
Add recommendations for collections

cc @cidrblock

### CODE REVIEW

**Review by ganeshrn — APPROVED** [review]

**Comment by ericzolf** [comment]
Rather straight forward statements, no issue with me. Just be more opinionated, less "should".

**Comment by tima** [comment]
This guidance seems thin especially relative to other sections. I'm not exactly sure what is missing, but seeing how little there is and knowing how big and important collections are gives me pause.

**Comment by jillr** [comment]
@tima Can you think of other suggestions that are collection specific and don't belong in other sections of the guide?  We can always open more PRs in the future but this was everything we had gathered so far that I didn't feel were more appropriate to a different heading in this repo.

File: collections/README.adoc [hunk 1]
```diff
@@ -49,3 +49,25 @@ beta_no_log: "{{ mycollection_no_log | default('false') }}"
...
 ====
 
+=== Collection Documentation
+[%collapsible]
+A collection should include a README file that is in the root of the collection and which contains:
+* Information about the purpose of the collection
+* A link to the collection license file
+* General usage information such as which versions of ansible-core are supported and any libraries or SDKs which are required by the collection
+
+Plugin documentation should be generated from DOCUMENTATION block in the plugin code.
```
Comment (mophahr) [inline collections/README.adoc:59]:
```suggestion
Plugin documentation should be generated from the DOCUMENTATION block in the plugin code.
```

File: collections/README.adoc [hunk 2]
```diff
@@ -49,3 +49,25 @@ beta_no_log: "{{ mycollection_no_log | default('false') }}"
...
++
+[collapsible]
+Examples::
+Use https://github.com/ansible-network/collection_prep to generate the documentation for the collection
++
+* Supplemental documentation such as user guides may be written in ReStructured Text (rst) and located in the docs/docsite/rst/ directory of the collection.
```
Comment (mophahr) [inline collections/README.adoc:65]:
```suggestion
* Supplemental documentation such as user guides may be written in reStructured Text (rst) and located in the docs/docsite/rst/ directory of the collection.
```

**Comment by jillr** [comment]
@mophahr I believe all comments have been addressed and this should be ready for re-review

**Comment by mophahr** [comment]
I took the liberty of fixing the remaining asciidoc issues @jillr

**Review by mophahr — APPROVED** [review]

**Comment by ericzolf** [comment]
Approved in CoP meeting 2022-06-29

**Pull request merged**